src/services/DexService.ts

```diff
@@ -35,61 +36,44 @@ export default class DexService {
-
-      addPropertyToClass(abilitiesData, 'first', page.item.abilities.first, requestedFields as GraphQLSet<keyof AbilitiesEntry>, 'abilities.first');
-      addPropertyToClass(
-        abilitiesData,
-        'second',
-        page.item.abilities.second,
-        requestedFields as GraphQLSet<keyof AbilitiesEntry>,
-        'abilities.second'
-      );
-      addPropertyToClass(
-        abilitiesData,
-        'hidden',
-        page.item.abilities.hidden,
-        requestedFields as GraphQLSet<keyof AbilitiesEntry>,
-        'abilities.hidden'
-      );
-      addPropertyToClass(
-        abilitiesData,
-        'special',
-        page.item.abilities.special,
-        requestedFields as GraphQLSet<keyof AbilitiesEntry>,
-        'abilities.special'
-      );
-
-      const dexEntryFields = requestedFields as GraphQLSet<keyof DexEntry>;
+      const genderEntryRequestedFields = Util.cast<GraphQLSet<keyof GenderEntry>>(requestedFields);
+      addPropertyToClass(genderData, 'male', `${pageGenderRatio.male * 100}%`, genderEntryRequestedFields, 'gender.male');
+      addPropertyToClass(genderData, 'female', `${pageGenderRatio.female * 100}%`, genderEntryRequestedFields, 'gender.female');
+
+      const baseStatsRequestedFields = Util.cast<GraphQLSet<keyof StatsEntry>>(requestedFields);
+      addPropertyToClass(baseStatsData, 'hp', page.item.baseStats.hp, baseStatsRequestedFields, 'baseStats.hp');
+      addPropertyToClass(baseStatsData, 'attack', page.item.baseStats.atk, baseStatsRequestedFields, 'baseStats.attack');
+      addPropertyToClass(baseStatsData, 'defense', page.item.baseStats.def, baseStatsRequestedFields, 'baseStats.defense');
+      addPropertyToClass(baseStatsData, 'specialattack', page.item.baseStats.spa, baseStatsRequestedFields, 'baseStats.specialattack');
+      addPropertyToClass(baseStatsData, 'specialdefense', page.item.baseStats.spd, baseStatsRequestedFields, 'baseStats.specialdefense');
+      addPropertyToClass(baseStatsData, 'speed', page.item.baseStats.spe, baseStatsRequestedFields, 'baseStats.speed');
+
+      const evYieldsRequestedFields = Util.cast<GraphQLSet<keyof EvYieldsEntry>>(requestedFields);
+      addPropertyToClass(evYieldsData, 'hp', page.item.evYields?.hp ?? 0, evYieldsRequestedFields, 'evYields.hp');
+      addPropertyToClass(evYieldsData, 'attack', page.item.evYields?.atk ?? 0, evYieldsRequestedFields, 'evYields.attack');
+      addPropertyToClass(evYieldsData, 'defense', page.item.evYields?.def ?? 0, evYieldsRequestedFields, 'evYields.defense');
+      addPropertyToClass(evYieldsData, 'specialattack', page.item.evYields?.spa ?? 0, evYieldsRequestedFields, 'evYields.specialattack');
+      addPropertyToClass(evYieldsData, 'specialdefense', page.item.evYields?.spd ?? 0, evYieldsRequestedFields, 'evYields.specialdefense');
+      addPropertyToClass(evYieldsData, 'speed', page.item.evYields?.spe ?? 0, evYieldsRequestedFields, 'evYields.speed');
```

Comment:
```suggestion
      addPropertyToClass(evYieldsData, 'hp', page.item.evYields?.hp ?? 0, evYieldsRequestedFields,       addPropertyToClass(evYieldsData, 'hp', page.item.evYields.hp, evYieldsRequestedFields, 'evYields.hp');
      addPropertyToClass(evYieldsData, 'attack', page.item.evYields.atk, evYieldsRequestedFields, 'evYields.attack');
      addPropertyToClass(evYieldsData, 'defense', page.item.evYields.def, evYieldsRequestedFields, 'evYields.defense');
      addPropertyToClass(evYieldsData, 'specialattack', page.item.evYields.spa, evYieldsRequestedFields, 'evYields.specialattack');
      addPropertyToClass(evYieldsData, 'specialdefense', page.item.evYields.spd, evYieldsRequestedFields, 'evYields.specialdefense');
      addPropertyToClass(evYieldsData, 'speed', page.item.evYields.spe, evYieldsRequestedFields, 'evYields.speed');
```

Once `evYields` has been added everywhere and the `?` in the `pokemon.d.ts` has been removed this change needs to be applied as well